tests/Unit/CustomerTest.php

```diff
@@ -366,4 +403,108 @@ public function testCreateCustomersContact()
+    public function testCreateCustomerNote()
+    {
+        $stream = Psr7\stream_for(CustomerTest::NOTE_JSON);
+        list($cmClient, $mockClient) = $this->getMockClient(0, [200], $stream);
+
+        $uuid = "cus_00000000-0000-0000-0000-000000000000";
+
+        $result = (new Customer(["uuid" => $uuid], $cmClient))->createCustomerNote(
+            [
+            "type" => "note",
+            "author_email" => "john@example.com",
+            "text" => "This is a note",
+            ]
+        );
+        $request = $mockClient->getRequests()[0];
+
+        $this->assertEquals("POST", $request->getMethod());
+        $uri = $request->getUri();
+        $this->assertEquals("/v1/customers/".$uuid."/notes", $uri->getPath());
+
+        $this->assertTrue($result instanceof CustomerNote);
+        $this->assertEquals("note_00000000-0000-0000-0000-000000000000", $result->uuid);
+    }
+
+    public function testUpdateCustomerNote()
+    {
+        $stream = Psr7\stream_for(CustomerTest::UPDATED_NOTE_JSON);
+        list($cmClient, $mockClient) = $this->getMockClient(0, [200], $stream);
+
+        $uuid = "cus_00000000-0000-0000-0000-000000000000";
+        $note_uuid = "note_00000000-0000-0000-0000-000000000000";
+
+        $result = (new Customer(["uuid" => $uuid], $cmClient))->updateCustomerNote(
+            $note_uuid,
+            [
+            "text" => "This is a new note",
+            ]
+        );
+        $request = $mockClient->getRequests()[0];
+
+        $this->assertEquals("PATCH", $request->getMethod());
+        $uri = $request->getUri();
+        $this->assertEquals("/v1/customers/".$uuid."/notes/".$note_uuid, $uri->getPath());
+        $this->assertEquals("This is a new note", $result->text);
+    }
+
+    public function testRetrieveCustomerNote()
+    {
+        $stream = Psr7\stream_for(CustomerTest::NOTE_JSON);
+        list($cmClient, $mockClient) = $this->getMockClient(0, [200], $stream);
+
+        $uuid = "cus_00000000-0000-0000-0000-000000000000";
+        $note_uuid = "note_00000000-0000-0000-0000-000000000000";
+
+        $result = (new Customer(["uuid" => $uuid], $cmClient))->retrieveCustomerNote(
+            $note_uuid
+        );
+        $request = $mockClient->getRequests()[0];
+
+        $this->assertEquals("GET", $request->getMethod());
+        $uri = $request->getUri();
+        $this->assertEquals("/v1/customers/".$uuid."/notes/".$note_uuid, $uri->getPath());
+
+        $this->assertTrue($result instanceof CustomerNote);
+        $this->assertEquals("note_00000000-0000-0000-0000-000000000000", $result->uuid);
+    }
+
+    public function testDeleteCustomerNote()
+    {
+        $stream = Psr7\stream_for('{}');
+        list($cmClient, $mockClient) = $this->getMockClient(0, [204], $stream);
+
+        $uuid = "cus_00000000-0000-0000-0000-000000000000";
+        $note_uuid = "note_00000000-0000-0000-0000-000000000000";
+
+        $result = (new Customer(["uuid" => $uuid], $cmClient))->deleteCustomerNote(
```

Comment:
If I understand it correctly, we should assert the $result for this:

https://github.com/chartmogul/chartmogul-php/blob/f74a8a506d91fb4e533566369de617bf8dc4ffd0/tests/Unit/CustomerTest.php#L495